### PR TITLE
viking: 1.7 -> 1.8

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
[Changelog](https://github.com/viking-gps/viking/releases/tag/viking-1.8)

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/viking/default.nix [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
+    ++ stdenv.lib.optional withMBTiles sqlite;
 
-  configureFlags = [ "--disable-scrollkeeper --disable-mapnik" ];
+  configureFlags = [
+    "--disable-scrollkeeper"
+  ] ++ stdenv.lib.optional (!withMapnik) "--disable-mapnik"
+    ++ stdenv.lib.optional (!withGeoClue) "--disable-geoclue"
+    ++ stdenv.lib.optional (!withMd5Hash) "--disable-nettle"
+    ++ stdenv.lib.optional (!withOAuth) "--disable-oauth"
+    ++ stdenv.lib.optional (!withMBTiles) "--disable-mbtiles";
```
Comment (marsam) [inline pkgs/applications/misc/viking/default.nix]:
```
  configureFlags = [
    "--disable-scrollkeeper"
    (stdenv.lib.enableFeature withMapnik "mapnik")
    (stdenv.lib.enableFeature withGeoClue "geoclue")
    (stdenv.lib.enableFeature withMd5Hash "nettle")
    (stdenv.lib.enableFeature withOAuth "oauth")
    (stdenv.lib.enableFeature withMBTiles "mbtiles")
  ];
```

File: pkgs/applications/misc/viking/default.nix [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
-  version = "1.7";
+  version = "1.8";
 
   src = fetchurl {
     url = "mirror://sourceforge/viking/viking/viking-${version}.tar.bz2";
-    sha256 = "092q2dv0rcz12nh2js1z1ralib1553dmzy9pdrvz9nv2vf61wybw";
+    sha256 = "1a0g0fbj4q5s9p8fv0mqvxws10q3naj81l72sz30vvqpbz6vqp45";
   };
 
+  patches = [ ./0001-fix-build.patch ];
```
Comment (marsam) [inline pkgs/applications/misc/viking/default.nix]: please use `fetchpatch` here:
```suggestion
  patches = [
    # Fix build without mapnik and sqlite https://github.com/viking-gps/viking/pull/79
    (fetchpatch {
      url = "https://github.com/viking-gps/viking/commit/995feefcb97bdb1590ed018224cf47ce197fe0c1.patch";
      sha256 = "1xb0b76kg690fag9mw3yfj5k766jmqp1sm8q4f29n1h3nz5g8izd";
    })
  ];
```

File: pkgs/applications/misc/viking/default.nix (hunk 1 above)
Comment (sikmir) [reply pkgs/applications/misc/viking/default.nix]: Fixed.

File: pkgs/applications/misc/viking/default.nix (hunk 2 above)
Comment (sikmir) [reply pkgs/applications/misc/viking/default.nix]: Fixed.

**Comment by marsam** [comment]
Thanks!

**Pull request merged**